turbinia/evidence.py

```diff
@@ -604,18 +604,22 @@ def _preprocess(self, _, required_states):
     # We need to enumerate partitions in preprocessing so the path_specs match
     # the parent evidence location for each task.
     try:
-      path_specs = partitions.Enumerate(self.parent_evidence)
-      path_specs_dicts = [path_spec.CopyToDict() for path_spec in path_specs]
-      log.debug(
-          'Enumerated path_specs {0!s} for parent Evidence {1:s}'.format(
-              path_specs_dicts, self.parent_evidence.name))
+      # We should only get one path_spec here since we're specifying the location.
+      path_specs = partitions.Enumerate(
+          self.parent_evidence, self.partition_location)
     except TurbiniaException as e:
       log.error(e)
 
-    path_spec = partitions.GetPathSpecByLocation(
-        path_specs, self.partition_location)
-    if path_spec:
-      self.path_spec = path_spec
+    if len(path_specs) > 1:
+      path_specs_dicts = [path_spec.CopyToDict() for path_spec in path_specs]
+      log.warning(
+          'Found more than one path_spec for {0:s}: {1!s}'.format(
+              self.parent_evidence.name, path_specs_dicts))
```

Comment:
So we don't want to try to keep `GetPathSpecByLocation` for this condition of multiple path_specs being returned?  Should we bail out with an exception at this point if we aren't going to search for the exact path_spec?